From 4887899da202ff6863609a29f851c52a2d49b022 Mon Sep 17 00:00:00 2001
From: "zhi.zhou" <zhi.zhou@amlogic.com>
Date: Fri, 25 Oct 2013 18:05:02 +0800
Subject: [PATCH 1594/5965] pd#81367 too many reset on wifi playing.ignore the
 first write block on rmvb,becuase maybe some data have no av data

---
 drivers/amlogic/amports/rmparser.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/rmparser.c b/drivers/amlogic/amports/rmparser.c
index 97efce12da38..57d7d2f87561 100755
--- a/drivers/amlogic/amports/rmparser.c
+++ b/drivers/amlogic/amports/rmparser.c
@@ -223,7 +223,13 @@ static ssize_t _rmparser_write(const char __user *buf, size_t count)
                 WRITE_MPEG_REG(PARSER_CONTROL, (ES_SEARCH | ES_PARSER_START));
                 printk("reset parse_control=%x\n",READ_MPEG_REG(PARSER_CONTROL));
             }
-            return -EAGAIN;
+            if(parse_halt <= 10){/*drops first 10 pkt ,some times maybe no av data*/
+				 printk("drop this pkt=%d,len=%d\n",parse_halt,len);
+                p += len;
+                r -= len;
+            }else{
+                return -EAGAIN;
+            }
         }else{
             parse_halt = 0;
             p += len;
-- 
2.19.0

